xbmc/GUIInfoManager.cpp

```diff
@@ -215,64 +217,86 @@ const infomap string_bools[] =   {{ "isempty",          STRING_IS_EMPTY },
                                   { "endswith",         STRING_ENDS_WITH },
                                   { "contains",         STRING_CONTAINS }};
 
-
 /// \page modules__infolabels_boolean_conditions
 /// \subsection modules__infolabels_boolean_conditions_Integer Integer
 /// \table_start
 ///   \table_h3{ Labels, Type, Description }
+///   \table_row3{   <b>`Integer.ValueOf(number)`</b>,
+///                  \anchor Integer_ValueOf
+///                  _integer_,
+///     @return An integer info label that represents the provided number
+///     @param number - the number to compute
+///     @note **Example:** `Integer.ValueOf(4)` will be evaluated to 4.
+///     @note Will return -1 if not able to convert the provided value to integer. **Example**: `Integer.ValueOf(some string)` will evaluate to -1
```

Comment:
I question the idea of returning -1. This makes sense for a programmer, but what does -1 mean for the user in the GUI? I would return 0 or an empty string

Reply:
Or add a fallback like Python does:

```
Integer.ValueOf(4, 0) will return 0 if not able to convert the provided value to an integer.
```

Just an idea

Reply:
(also notice the missing `an` if you stick with -1)

Reply:
@garbear this is an integer info not a string one (infolabel). The value is not shown to the user anywhere as a label but rather used internally to convey the fixed number values in integer comparisons so that internally we always register and compare info expressions. I.e.:
`Integer.GreaterThan(info, 4)` is the same as `Integer.GreaterThan(info, Integer.ValueOf(4))`. Note that on the first one "4" itself is a string.

It's documented because by making use of it we end up also exposing it (and does not harm anyway to have it available).
I also don't like the -1 but returning 0 is pretty much the same here imho. Integer.ValueOf(test) simply can't be evaluated to an integer. As there is no way to express invalid data in an integer value I thought -1 would be the best representation possible.

Reply:
I see what you mean in the value not being exposed to the user. And agree that -1 is a fine value for invalid data. Though I would use 0 because it's a more natural number. Your call.

Reply:
and don't forget the article word: it should be "to _an_ integer"